### PR TITLE
Improve OIDC bearer token concept doc

### LINKED ISSUE #37463: Docs: security-oidc-bearer-token-authentication guide
### Describe the bug

Going through the [security-oidc-bearer-token-authentication](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication) guide I found some issues.
1. Links:
  \- `OIDC authorization code flow authentication` should be https://quarkus.io/version/main/guides/security-oidc-code-flow-authentication-tutorial
  \- Reference `Protect service applications by using OIDC Bearer token authentication` is there 2 times and one link show 404
2. It would be nice to mention that some of the snippets are from quickstart or Quarkus tests. As this can be for user nicer to just look if they are not sure what all should be set. I will mention this in some points in this issue. My point on this is that if we have it somewhere in Quarkus managed repository it can be linked.  And part of this point is that it can be said here is quickstart and you can extend by some of these snippets.
3. [Token Claims And SecurityIdentity Roles
](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#token-claims-and-security-identity-roles) Would be nice to have some showcase for example with existing quickstart. How would the `application.properties` would look with token claims.
4. [Token scopes And SecurityIdentity permissions](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#token-scopes-and-security-identity-permissions) code snippet missing the import of List. Also missing where is the Order class , how it should look. Search it in `quarkusio` organisation and didn't find anything. [Search](https://github.com/search?q=org%3Aquarkusio+%22listOrders%22&type=code). Maybe it's not needed in this snippet at all.
5. [JSON Web Token Claim Verification](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#jwt-claim-verification) snippet missing import of `jakarta.ws.rs.core.Response`
6. [Single page applications](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#single-page-applications) snippet no longer work with keycloak 22+ see [this](https://github.com/keycloak/keycloak/blob/main/docs/documentation/release_notes/topics/22_0_0.adoc#legacy-promise-api-removed)
7. [Testing](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#integration-testing) In general I would move part of the testing to [security-oidc-bearer-token-authentication-tutorial](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication-tutorial) as some of the showcase is part of the solution for that tutorial ([quickstart for it](https://github.com/quarkusio/quarkus-quickstarts/tree/main/security-openid-connect-quickstart/src/test/java/org/acme/security/openid/connect)). Specifically  [Dev Services for Keycloak](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#integration-testing-keycloak-devservices) section. Think it base on this quickstart.
8. [Wiremock](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#integration-testing-wiremock)
\- `BearerTokenAuthorizationTest` missing static import `org.hamcrest.Matchers.equalTo`.
\- In this section can be said that it can extended the quickstart with some modification as `application.properties`
10. [OidcTestClient](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#integration-testing-oidc-test-client) missing definition of `client`. Probably it should be `oidcTestClient`. After that modification I have been unable to still run this test but probably some misconfiguration on my side. Error was `io.vertx.core.json.DecodeException: Failed to decode:Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')`. 
11. [KeycloakTestResourceLifecycleManager](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#integration-testing-keycloak)
\- Unused import `org.hamcrest.Matchers`. 
\- The `RestAssured.given().auth().oauth2(getAccessToken("alice"))))` have more bracket then expected should be 2 less bracket at the end.
\- how should be `/api/users/preferredUserName` setup. There would be nice to have link to the [UserResource](https://github.com/quarkusio/quarkus/blob/main/integration-tests/oidc/src/main/java/io/quarkus/it/keycloak/UsersResource.java) where it is setup.
12. [Local Public Key](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#integration-testing-public-key) for showcase purpose it would be nice to add link to [privateKey.pem](https://github.com/quarkusio/quarkus/tree/main/integration-tests/oidc-tenancy/src/main/resources) as it say to get it from `integration-tests/oidc-tenancy`.
13. [TestSecurity annotation](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#integration-testing-security-annotation)
\- In `TestSecurityAuthTest` snippet is import `io.quarkus.test.security.oidc.ConfigMetadata`. It's not exist, looking on older Quarkus version probably this never exist + this import is never used in snippet. 
\- Both `ProtectedResource` snippets missing some imports [1].
14. [Authentication after HTTP request has completed](https://quarkus.io/version/main/guides/security-oidc-bearer-token-authentication#authentication-after-http-request-has-completed)
\- `OrderResource` missing `jakarta.ws.rs.HeaderParam` import
\- In both snippets is Product class. Not exactly know what structure should this class have. It should be [this one](https://github.com/quarkusio/quarkus/blob/main/integration-tests/oidc-wiremock/src/main/java/io/quarkus/it/keycloak/Product.java) so maybe some link to this class or link to this set of test for more info would be nice.
\- `OrderService` missing the variable name for `SecurityIdentity`.

Personally thinking that the import should be there or the snippets shouldn't have any imports. Missing them is not a big deal but mention them here. For the point number 6. there should be probably decision which keycloak to mention in docs as in some guides is mention keycloak 17+ which will be working but for example the point 6 won't be working.

[1] Missing imports
```
import io.quarkus.security.Authenticated;
import jakarta.inject.Inject;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #37463

SPA snippet currently does not work with KC 23.
CC @jedla97, we can finalize early next year if Jakub is on PTO

### CODE REVIEW

**Comment by rsvoboda** [comment]
Jakub added his comments to the original issue - https://github.com/quarkusio/quarkus/issues/37463#issuecomment-1852387025

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc [hunk 1]
```diff
@@ -782,7 +818,7 @@ quarkus.oidc.public-key=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAlivFI8qB4D0y
 smallrye.jwt.sign.key.location=/privateKey.pem
 ----
 
-copy `privateKey.pem` from the `integration-tests/oidc-tenancy` in the `main` Quarkus repository and use a test code similar to the one in the `Wiremock` section above to generate JWT tokens. You can use your own test keys if preferred.
+copy link:https://github.com/quarkusio/quarkus/tree/main/integration-tests/oidc-tenancy/src/main/resources/privateKey.pem[privateKey.pem] from the `integration-tests/oidc-tenancy` in the `main` Quarkus repository and use a test code similar to the one in the `Wiremock` section above to generate JWT tokens. You can use your own test keys if preferred.
```
Comment (jedla97) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:821]: Thanks for adding the link. Just small note it should look something like `link:{quickstarts-tree-url}/integration-tests/oidc-tenancy/src/main/resources/privateKey.pem[privateKey.pem]`

**Review by jedla97 — CHANGES_REQUESTED** [review]
Sorry I miss the push for this PR. Thanks for all the changes! Looking at the changes they look good just one comment. Also just one thing is that text `If you want to protect web applications by using OIDC authorization code flow authentication, see OIDC authorization code flow authentication` still point to 404 and by your comment (point 1) https://github.com/quarkusio/quarkus/issues/37463#issuecomment-1852238946 this should be also removed.

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc (hunk 1 above)
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:821]: @jedla97 Unfortunately this resource is only available in the main repository, not in quickstarts. This doc is a concept/reference doc, so it is not strictly associated with the quickstart

**Comment by sberyozkin** [comment]
@jedla97 Thanks, I think I've fixed the link, now it should correctly point to the code flow concept/ref doc, `-concept` is no longer part of the file name

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc (hunk 1 above)
Comment (jedla97) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:821]: Oh sorry how i was jumping between guides I miss that this link to Quarkus repo. What I had tried now is this `link:{quarkus-blob-url}/integration-tests/oidc-tenancy/src/main/resources/privateKey.pem[privateKey.pem]` and it should work.

**Review by jedla97 — APPROVED** [review]
I can confirm that 404 link is gone. For that one hardcoded link (previous comment) I don't fully mind it can stay as it. So all my concerns and problems from issue was fixed or explained. With this I can approve this PR.

**Review by rsvoboda — APPROVED** [review]
Didn't spot anything suspicious, approving based on the approval from Jakub.

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc (hunk 1 above)
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:821]: Thanks @jedla97 Sure, let me do it

**Comment by sberyozkin** [comment]
Thanks @jedla97 @rsvoboda, did one more update as suggested by @jedla97, should be ready to go once the build is finished

**Review by jedla97 — APPROVED** [review]
Thanks for updating link. Tested it and it's work.

**Pull request merged**